### PR TITLE
Important iconv fixes + test

### PR DESCRIPTION
**Pull request by Lyude**
Realized that the code I had written for gskill to make sure there was always a profile name wasn't working like I had thought it was… traced down the issue to being that the iconv helper was not working anywhere as well as I thought it was and figured in addition to fixing that it would be a good idea to write some basic tests to make sure it doesn't break in the future.

### CODE REVIEW

File: test/test-iconv-helper.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+int main(void)
+{
+	int nfailed;
+	Suite *s;
+	SRunner *sr;
+
+	s = test_context_suite();
+        sr = srunner_create(s);
```
Comment (whot) [inline test/test-iconv-helper.c]: indentation

File: test/test-iconv-helper.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2015 Red Hat, Inc.
```
Comment (whot) [inline test/test-iconv-helper.c]: your calendar is running slow :P

File: test/test-iconv-helper.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	char output[10] = { 0 };
+	ssize_t rc;
+
+	rc = ratbag_utf8_to_enc(output, sizeof(output), "Enterprise UTF-16LE",
+				"%s", "Foo");
+
+	ck_assert_int_eq(rc, -EINVAL);
+}
+END_TEST
+
```
Comment (whot) [inline test/test-iconv-helper.c]: I'm missing a test here for { 'F', '\0' } (i.e. a single-byte string) and tests for other odd-numbered byte strings

also, what happens when you have multiple consecutive '\0' in the source string?

**Comment by whot** [comment]
so, I clearly don't know what I'm doing and assumed that the first "ok" applied only to that commit rather than the pull request. Please fix up the couple of nitpicks and re-submit, thanks

File: test/test-iconv-helper.c (hunk 3 above)
Comment (Lyude) [inline test/test-iconv-helper.c]: Sorry for the duplicate, still getting the hang of this thing…
If you're talking about something like

```c
ratbag_utf8_to_enc(output, sizeof(output), "UTF-16LE", "%s", "F\0\0\0oo");
```
nothing really interesting would happen. `vsnprintf()` would see the first `'\0'` and assume the entirety of the string is `"F"`

**Comment by Lyude** [comment]
Are there still any pending review changes for this?

**Pull request merged**